src/idsk/components/stepper/stepper.js

```diff
@@ -0,0 +1,288 @@
+  this.openAllClass = 'idsk-stepper__open-all'
+  this.iconClass = 'idsk-stepper__icon'
+
+  this.sectionHeaderClass = 'idsk-stepper__section-header'
+  this.sectionHeaderFocusedClass = 'idsk-stepper__section-header--focused'
+  this.sectionHeadingClass = 'idsk-stepper__section-heading'
+  this.sectionSummaryClass = 'idsk-stepper__section-summary'
+  this.sectionButtonClass = 'idsk-stepper__section-button'
+  this.sectionExpandedClass = 'idsk-stepper__section--expanded'
+}
+
+// Initialize component
+Stepper.prototype.init = function () {
+  // Check for module
+  if (!this.$module) {
+    return
+  }
+
+  this.initControls()
+  this.initSectionHeaders()
+
+  nodeListForEach(
+    this.$links,
+    function ($link) {
+      $link.addEventListener('focus', this.handleItemLink.bind(this));
+      $link.addEventListener('blur', this.handleItemLink.bind(this));
+    }.bind(this)
+  );
+
+  // See if "Zobraziť všetko" button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+// Initialise controls and set attributes
+Stepper.prototype.initControls = function () {
+  // Create "Zobraziť všetko" button and set attributes
+  this.$openAllButton = document.createElement('button')
+  this.$openAllButton.setAttribute('type', 'button')
+  this.$openAllButton.innerHTML = 'Zobraziť všetko <span class="govuk-visually-hidden">sections</span>'
+  this.$openAllButton.setAttribute('class', this.openAllClass)
+  this.$openAllButton.setAttribute('aria-expanded', 'false')
+  this.$openAllButton.setAttribute('type', 'button')
+
+  // Create control wrapper and add controls to it
+  var accordionControls = this.$module.querySelector('.idsk-stepper__controls');
+  accordionControls.appendChild(this.$openAllButton)
+   
+  // Handle events for the controls
+  this.$openAllButton.addEventListener('click', this.onOpenOrCloseAllToggle.bind(this))
+}
+
+// Initialise section headers
+Stepper.prototype.initSectionHeaders = function () {
+  // Loop through section headers
+  nodeListForEach(this.$sections, function ($section, i) {
+    // Set header attributes
+    var header = $section.querySelector('.' + this.sectionHeaderClass)
+    this.initHeaderAttributes(header, i)
+
+    this.setExpanded(this.isExpanded($section), $section)
+
+    // Handle events
+    header.addEventListener('click', this.onSectionToggle.bind(this, $section))
+
+    // See if there is any state stored in sessionStorage and set the sections to
+    // open or closed.
+    this.setInitialState($section)
+  }.bind(this))
+}
+
+Stepper.prototype.handleItemLink = function (e) {
+  var $link = e.target || e.srcElement;
+  var $currentSection = $link.closest('.idsk-stepper__section');
+  toggleClass($currentSection, 'idsk-stepper__bolder-line');
+}
+
+
+// Set individual header attributes
+Stepper.prototype.initHeaderAttributes = function ($headerWrapper, index) {
+  var $module = this
```

Comment:
tu nemalo byt this.module? ale len this? tak isto je to aj v dalsich funkciach

Reply:
tu som to zmenil ale v jednej dalsej to takto musi byt kvoli tomu ze ten this potrebujem do foreachu vo funkcii